include/alp/type_traits.hpp

```diff
@@ -148,6 +185,71 @@ namespace alp {
 
 	} // end namespace alp::internal
 
+	/**
+	 * Used to get a structure type of the given ALP container
+	 *
+	 * @tparam T The ALP container to inspect.
+	 *
+	 */
+	template< typename Container >
+	struct inspect_structure {};
+
+	/**
+	 * Used to get a View type of the given ALP container
+	 *
+	 * @tparam T The ALP container to inspect.
+	 *
+	 */
+	template< typename Container >
+	struct inspect_view {};
```

Comment:
internal? Perhaps a user-level `is_view` could be OK. Not sure when useful but at least users are aware of the concept and might want to use such a trait. But I don't think a user should/would be able to process View types as they shouldn't be accessible to her.